src/modules/hip/kernel/resize.hpp

```diff
@@ -355,55 +355,80 @@ __global__ void resize_generic_pkd_tensor(T *srcPtr,
-    rpp_hip_compute_interpolation_scale_and_radius(interpolationType, &hScale, &hRadius, hRatio);
     float wOffset = (wRatio - 1) * 0.5f - wRadius;
     float hOffset = (hRatio - 1) * 0.5f - hRadius;
     int wKernelSize = ceilf(wRadius * 2);
     int hKernelSize = ceilf(hRadius * 2);
 
     float rowWeight, colWeight, rowCoeff, colCoeff;
+    __shared__ float rowCoeffs[45];
+    float colCoeffs[45];
+    __shared__ float rowCoeffSum;
+    rowCoeffSum = 0.0f;
+    float colCoeffSum = 0.0f;
+    float3 coeffs_f3[45] = {(float3)0.0f};
+
     int srcLocationRowFloor, srcLocationColumnFloor;
     resize_roi_generic_srcloc_and_weight_hip_compute(srcRoi_i4.x, id_x, wRatio, widthLimit, &srcLocationColumnFloor, &colWeight, wOffset, 3);
     resize_roi_generic_srcloc_and_weight_hip_compute(srcRoi_i4.y, id_y, hRatio, heightLimit, &srcLocationRowFloor, &rowWeight, hOffset, 1);
 
     T *srcPtrTemp = srcPtr + (id_z * srcStridesNH.x);
     float3 outPixel_f3 = (float3)0.0f;
-    float rowCoeffSum = 0.0f, colCoeffSum = 0.0f;
+    if(hipThreadIdx_x==0)
+    {
+        for(int j = 0; j < hKernelSize; j++)
+        {
+            rpp_hip_compute_interpolation_coefficient(interpolationType, (rowWeight - hRadius + j) * hScale , &rowCoeff);
+            rowCoeffSum += rowCoeff;
+            rowCoeffs[j] = rowCoeff;
+        }
+        rowCoeffSum = (rowCoeffSum == 0.0f) ? 1.0f : rowCoeffSum;
+    }
+    __syncthreads();
+
+    for(int k = 0; k < wKernelSize; k++)
+    {
+        rpp_hip_compute_interpolation_coefficient(interpolationType, (colWeight - wRadius + k) * wScale , &colCoeff);
+        colCoeffSum += colCoeff;
+        colCoeffs[k] = colCoeff;
+    }
+
+    colCoeffSum = (colCoeffSum == 0.0f) ? 1.0f : colCoeffSum;
+
     for(int j = 0; j < hKernelSize; j++)
     {
         int rowIndex = fminf(fmaxf((int)(srcLocationRowFloor + j), 0), heightLimit);
         T *srcRowPtrsForInterp = srcPtrTemp + rowIndex * srcStridesNH.y;
-        rpp_hip_compute_interpolation_coefficient(interpolationType, (rowWeight - hRadius + j) * hScale , &rowCoeff);
-        rowCoeffSum += rowCoeff;
 
-        colCoeffSum = 0;
         for(int k = 0; k < wKernelSize; k++)
         {
+            float3 src_f3;
             int colIndex = fminf(fmaxf((int)(srcLocationColumnFloor + (k * 3)), 0), widthLimit);
-            rpp_hip_compute_interpolation_coefficient(interpolationType, (colWeight - wRadius + k) * wScale , &colCoeff);
-            colCoeffSum += colCoeff;
-            float3 coeff_f3 = (float3)(colCoeff * rowCoeff);
-            outPixel_f3 += (make_float3(srcRowPtrsForInterp[colIndex], srcRowPtrsForInterp[colIndex + 1], srcRowPtrsForInterp[colIndex + 2]) * coeff_f3);
+            rpp_hip_interpolate3_triangular_load_pkd3(srcRowPtrsForInterp, colIndex, &src_f3); // To add helper function for I8 and float
+            coeffs_f3[k] +=  src_f3 * (float3)(rowCoeffs[j] / rowCoeffSum);
```

Comment:
Again loop independent can be taken outside here, divisions to be precomputed. We have discussed these. There is a lot of scope for improvement here on code standard